pkg/core/basic_cluster.go

```diff
@@ -97,7 +99,7 @@ type RegionSetInformer interface {
 	GetAdjacentRegions(region *RegionInfo) (*RegionInfo, *RegionInfo)
 	ScanRegions(startKey, endKey []byte, limit int) []*RegionInfo
 	GetRegionByKey(regionKey []byte) *RegionInfo
-	BatchScanRegions(keyRanges *KeyRanges, limit int) []*RegionInfo
+	BatchScanRegions(keyRanges *KeyRanges, limit int, outputMustContainAllKeyRange bool) ([]*RegionInfo, error)
```

Comment:
I hope it's 

```golang
BatchScanRegions(keyRanges *KeyRanges, limit int, scanOptions ...BatchScanRegionsOptions) ([]*RegionInfo, error)
```

So in future we don't need to change the signature.

Reply:
+1

Reply:
lgtm

Reply:
But this is an internal interface and it won't change the way it's called externally. @lance6716 

Grpc interface is here https://github.com/pingcap/kvproto/pull/1252

Reply:
For internal interfaces, I prefer to wait for more options to emerge before tweaking to avoid over-optimization. @rleungx

Reply:
Although it's an internal interface, if PD developers will add more options in future, using `...BatchScanRegionsOptions` can save your time of editing the signature of interface and implementation.

Reply:
option will help us reduce the modification of the interface